### PR TITLE
refactor(osmoutils): index position by string name instead of address

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Index position by string name instead of the address. This is required for some use cases such as concentrated liquidity (CL).

In CL, we need to index positions by the owner, pool id, and ticks.

The flexibility to index by address is still provided since address can easily be converted to string.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: osmoutils/accum/accum_helpers.go [hunk 1]
```diff
@@ -7,20 +7,20 @@ import (
 )
 
 // Creates a new position at accumulator's current value with a specific number of shares and unclaimed rewards
-func createNewPosition(accum AccumulatorObject, addr sdk.AccAddress, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
+func createNewPosition(name AccumulatorObject, index string, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
```
Comment (AlpinYukseloglu) [inline osmoutils/accum/accum_helpers.go:10]: Trying to reconcile this renaming with the other functions and tests – here you call the `AccumulatorObject` `name`, whereas in `AddToPosition` and in tests you call the address string `name`. We should probably align these before merging

**Review by AlpinYukseloglu — APPROVED** [review]
Left one comment about naming inconsistency but other than that LGTM

File: osmoutils/accum/accum_helpers.go (hunk 1 above)
Comment (czarcas7ic) [reply osmoutils/accum/accum_helpers.go:10]: Second this

**Review by czarcas7ic — APPROVED** [review]

File: osmoutils/accum/accum_helpers.go (hunk 1 above)
Comment (p0mvn) [reply osmoutils/accum/accum_helpers.go]: Erroneous change. Fixed this and other name issues found while re-reviewing

**Pull request merged**